src/lib/components/VideoCard/VideoCard.svelte

```diff
@@ -1,78 +1,154 @@
 <script lang="ts">
   import "./VideoCard.scss";
 
-  import { afterUpdate } from "svelte";
-
-  import getYoutubeVideoData from "$lib/services/youtube";
+  import type { Thumbnails } from "$lib/services/server/youtube";
+  import getYoutubeVideoIDFromURL from "$lib/util/getYoutubeVideoIDFromURL";
   import YoutubeSubscribeLink from "$lib/components/YoutubeSubscribeLink";
+  import Image, { type Sources, type Srcset } from "$lib/components/Image";
 
   type VideoCardVariation = "hero" | "primary" | "secondary" | "tertiary";
   interface $$Props {
     class?: string;
     url: string;
-    customTitle?: string | null | undefined;
-    customDescription?: string | null | undefined;
+    title?: string | null | undefined;
+    description?: string | null | undefined;
     variation?: VideoCardVariation;
+    blurhash?: string | null | undefined;
+    thumbnails?: Thumbnails;
+    sizeType?: SizeType;
   }
 
   let className = "";
   export { className as class };
   export let url = "";
-  export let customTitle: $$Props["customTitle"] = null;
-  export let customDescription: $$Props["customDescription"] = null;
+  export let title: $$Props["title"] = null;
+  export let description: $$Props["description"] = null;
   export let variation = "primary";
 
-  let youtubeVideoId: string | null;
+  const maxDescriptionLength = 500;
+  $: trimmedDescription = description
+    ?.replace(/\n.+$/, "\n")
+    .slice(0, maxDescriptionLength)
+    .replace(/\s[^\s]*?$/, "");
```

Comment:
Tested with a screen reader and the entire description gets read, regardless of how much is actually showing on the screen. On the Animals top tier page the description gets cut-off mid-sentence, so we should probably try to break on the first paragraph since that would be easier than trying to determine where one sentence ends and another begins.

Reply:
This code was _supposed_ to break at the first paragraph _or_ 500 characters (while preventing mid-word breaks), whichever was smaller. Unfortunately I messed up the regex a little bit. I've pushed a fix.